suse_migration_services/units/migrate.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2018 SUSE Linux LLC.  All rights reserved.
+#
+# This file is part of suse-migration-services.
+#
+# suse-migration-services is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# suse-migration-services is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with suse-migration-services. If not, see <http://www.gnu.org/licenses/>
+#
+import yaml
+
+# project
+from suse_migration_services.command import Command
+from suse_migration_services.defaults import Defaults
+
+from suse_migration_services.exceptions import (
+    DistMigrationZypperException
+)
+
+
+def main():
+    """
+    DistMigration run zypper based migration
+
+    Call zypper migration plugin and migrate the system
+    """
+    root_path = Defaults.get_system_root_path()
+
+    try:
+        Command.run(
+            [
+                'zypper', 'migration',
+                '--gpg-auto-import-keys',
+                '--no-selfupdate',
+                '--migration', get_product_selection_index(),
```

Comment:
I think we want to drop the use of the "--migration" option and add "--non-interactive".

Only when "--non-interactive" is set  for migration will the option be passed to zypper [1]. When "--non-interactive" is set and "--migration" is not set then the migration plugin handles the "number based selection" [2].

This approach will get us out of the business of picking a number. If the meaning of the numbered selection ever changes this implementation will not be broken as the internal implementation of what

`migration_num = options[:migration]
if options[:non_interactive] && migration_num == 0
  # select the first option
  migration_num = 1
end`

means would also have to change.

[1] https://github.com/SUSE/zypper-migration/blob/master/zypper-migration#L519
[2] https://github.com/SUSE/zypper-migration/blob/master/zypper-migration#L397

Reply:
I also think it would be good to change the zypper-migration code to reflect this. You might have noticed that @jesusbv created a pull request against the zypper-migration repo with some adaptions.

@jesusbv do you think you can extend your changes to let us call the migration independent of any selection index ?

Reply:
I think code wise there would be no change but no include --migration option in the zypper migration call. I guess in this case, it is known that it is going to be the first option.